api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java

```diff
@@ -127,6 +133,11 @@ public org.openmrs.Encounter toOpenmrsType(@Nonnull org.openmrs.Encounter existi
 		        .getParticipant().stream().map(encounterParticipantComponent -> participantTranslator
 		                .toOpenmrsType(new EncounterProvider(), encounterParticipantComponent))
 		        .collect(Collectors.toCollection(LinkedHashSet::new)));
+
+		for (EncounterProvider ep : existingEncounter.getEncounterProviders()) {
+			ep.setEncounter(existingEncounter);
+			ep.setEncounterRole(getDefaultEncounterRole());
+		}
```

Comment:
Instead of doing this, could we include an additional `.map()` step in the above that does the same?